apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-launch.js

```diff
@@ -64,7 +61,7 @@ function WelcomeTour() {
 	const { setJustMaximized } = useWelcomeTourContext();
 
 	// Preload card images
-	tourSteps.forEach( ( step ) => ( new window.Image().src = step.meta.imgSrc ) );
+	tourSteps.forEach( ( step ) => <PrefetchImage asset={ step.meta.imgSrc } /> );
```

Comment:
I feel we may be introducing a bit of an anti-pattern with calling JSX components in this fashion 🤔

Sorry if I implied something of this nature in my comment. Have we considered implementing a hook for this, something like `useImagePrefetcher`, and do the looping there maybe? So a signature of the form `( assets: TourAsset[] ) => void`, and see if also usable for the optimisation we are removing [above](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-launch.js#L31)?

Thanks for looking into this.

Reply:
> I feel we may be introducing a bit of an anti-pattern with calling JSX components in this fashion 🤔

I agree. JSX should be used to render components and not to prefetch images (although, IMHO, it's quite cool that it's even possible!), hence a hook is a better solution. I've changed it in the latest commit.